### DISCUSSION TITLE
Permission denied to set role 'admin'

### DISCUSSION THREAD

**Discussion by ChrisRamirez11** [supabase/supabase#29864] [Questions] [answered]
I set as admin an user registered, in Supabase it says that he is admin but in my app when I get currentUser.role it says authenticated. Then when I sign out and sign in again now it says admin, and then the error permission denied to set role admin

**Comment by GaryAustin1** [comment, marked as answer]
Are you using a guide for this?
How are you setting the role claim?
Also you can't just change the role claim to anything.  There has to be a Postgres role defined with that name and granted access to the authenticator role and granted access to the database schemas.

In general using a postgres role and the role claim is not the recommended way to do custom roles.  Here is one guide from Supabase on roles (but not using the role claim).  https://supabase.com/docs/guides/database/postgres/custom-claims-and-role-based-access-control-rbac


Link with a bit of info on doing the Postgres role approach: https://supabase.com/docs/guides/storage/schema/custom-roles
